sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/ScalaReflection.scala

```diff
@@ -306,7 +330,7 @@ object ScalaReflection extends ScalaReflection {
    * input object is located at ordinal 0 of a row, i.e., `BoundReference(0, _)`.
    */
   def serializerFor(enc: AgnosticEncoder[_]): Expression = {
```

Comment:
TODO check the generated code for boxed primitives. We might be doing double conversions there.